### ISSUE TITLE
Use of the riak_dt_map is unsafe.

### ISSUE THREAD

**Issue by cmeiklejohn** [cmeiklejohn/riak_pg#8]
@marksteele 

I believe that this code in the ```vnode``` is unsafe:

```
  Pids = proplists:get_value({Group, riak_dt_orswot}, riak_dt_map:value(Groups), []),
```

In the event one node is ahead by a delete, the FSM used to coordinate the request needs to merge maps before returning the result to the user.  In this case, the coordinating FSM can not which causally came first: the difference between a removal after addition, or failure to observe the addition.

I'd like to rever the change to the map and open a branch for working on it, if you don't disapprove of the change, until we sort out any race conditions or safety concerns with the map.

**Comment by marksteele** [comment]
I'll be updating the PR shortly, already fixed.

**Comment by cmeiklejohn** [comment]
Addressed by #7.
